### PR TITLE
PROGRAMMES-6505 Table Content Block

### PR DESCRIPTION
**Pull request by votemike**
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6505

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/ExternalApi/Isite/Mapper/ContentBlockMapper.php [hunk 1]
```diff
@@ -49,6 +50,57 @@ public function getDomainModel(SimpleXMLElement $isiteObject)
                     $links
                 );
                 break;
+            case 'table':
```
Comment (votemike) [inline src/ExternalApi/Isite/Mapper/ContentBlockMapper.php]: What would be the best way to test this?

File: src/Ds2013/Presenters/Domain/ContentBlock/Table/table.html.twig [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<div class="component component--box{{ table.isInPrimaryColumn() ? ' component--box-flushbody-vertical component--box--primary' : ' component--box--striped component--box--secondary' }}">
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Domain/ContentBlock/Table/table.html.twig]: How common is this logic across the different content block types. Is it, theoretically, worth creating a BaseContentBlockPresenter that does this logic everywhere, and just has getXClasses() methods that its children can call?

Comment (votemike) [reply src/Ds2013/Presenters/Domain/ContentBlock/Table/table.html.twig]: Not sure, in V2 it's all hidden in that component renderer. I have a feeling that the 2 blocks that have already been done for the profile page don't take in to account the secondary column. Maybe the person who picks up the next ticket can look in to genericising it.

File: src/ExternalApi/Isite/Mapper/ContentBlockMapper.php [hunk 3]
```diff
@@ -6,6 +6,7 @@
 use App\ExternalApi\Isite\Domain\ContentBlock\AbstractContentBlock;
```
Comment (Tasiobg) [inline src/ExternalApi/Isite/Mapper/ContentBlockMapper.php]: I think you can remove
use App\ExternalApi\Isite\Domain\ContentBlock\AbstractContentBlock;
use Exception;

Comment (votemike) [reply src/ExternalApi/Isite/Mapper/ContentBlockMapper.php]: Both of those classes will be used once all blocks have been implemented.

File: src/ExternalApi/Isite/Mapper/ContentBlockMapper.php (hunk 1 above)
Comment (votemike) [reply src/ExternalApi/Isite/Mapper/ContentBlockMapper.php]: Done

**Pull request merged**